resources/ebs_volume.rb

```diff
@@ -131,7 +131,11 @@
     old_snapshots[new_resource.snapshots_to_keep, old_snapshots.length].each do |die|
       converge_by("delete snapshot with id: #{die[:snapshot_id]}") do
         Chef::Log.info "Deleting old snapshot #{die[:snapshot_id]}"
-        ec2.delete_snapshot(snapshot_id: die[:snapshot_id])
+        begin
+          ec2.delete_snapshot(snapshot_id: die[:snapshot_id])
+        rescue Aws::EC2::Errors::InvalidSnapshotInUse # rubocop: disable Lint/HandleExceptions
+          # Snapshot is being used by an AMI and therefore cannot be deleted.
```

Comment:
Can you add a Chef::Log.debug message here stating that the snapshot and ID cannot be removed

Reply:
Thanks for the review. Updated and added the message. Let me know if it's ok like this.